internal/service/firehose/sweep.go

```diff
@@ -30,30 +29,30 @@ func sweepDeliveryStreams(region string) error {
 	input := &firehose.ListDeliveryStreamsInput{}
 	sweepResources := make([]*sweep.SweepResource, 0)
 
-	for {
-		page, err := conn.ListDeliveryStreams(input)
-
-		if sweep.SkipSweepError(err) {
-			log.Printf("[WARN] Skipping Kinesis Firehose Delivery Streams sweep for %s: %s", region, err)
-			return nil
-		}
-
-		if err != nil {
-			return fmt.Errorf("error listing Kinesis Firehose Delivery Streams: %w", err)
+	err = listDeliveryStreamsPages(conn, input, func(page *firehose.ListDeliveryStreamsOutput, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
 		}
 
-		for _, sn := range page.DeliveryStreamNames {
+		for _, v := range page.DeliveryStreamNames {
 			r := ResourceDeliveryStream()
 			d := r.Data(nil)
 			d.SetId("???")
```

Comment:
Setting the ID to `???` is not ideal, since it gives log messages like

> [INFO] While sweeping resource (???), encountered throttling error

Ideally, the name would have been used as the ID, since all API calls take the name. For readable logs, the ID should be set either to the name or the ARN should be retrieved.

Reply:
Now synthesizing the delivery stream ARN and using that for the resource ID.